### PR TITLE
update has algorithm to sha256 and add upgrade managers to cover upgrade flow

### PR DESCRIPTION
**Pull request by helenren**

### CODE REVIEW

File: SalesforceSDK/Universal/Hybrid/Logger.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
+﻿/*
+ * Copyright (c) 2015, salesforce.com, inc.
```
Comment (bhariharan) [inline SalesforceSDK/Universal/Hybrid/Logger.cs]: ```2016```.

File: SalesforceSDK/Universal/Upgrade/SDKUpgradeManager.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
+﻿using System;
```
Comment (bhariharan) [inline SalesforceSDK/Universal/Upgrade/SDKUpgradeManager.cs]: Add copyright.

File: SalesforceSDK/Universal/Upgrade/SDKUpgradeManager.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        public async Task UpgradeAsync()
+        {
+            if (ApplicationData.Current.Version.Equals(0))
+            {
+                await UpgradeFromEarlierThan4Dot2();
+            }
+            await ApplicationData.Current.SetVersionAsync(VersionConvertion(SDKManager.SDK_VERSION), VersionRequestHandler);
+        }
+
+        private async Task UpgradeFromEarlierThan4Dot2()
```
Comment (bhariharan) [inline SalesforceSDK/Universal/Upgrade/SDKUpgradeManager.cs]: Can we add tests for this?

**Comment by bhariharan** [comment]
Looks good.

File: SalesforceSDK/Universal/Upgrade/SDKUpgradeManager.cs (hunk 3 above)
Comment (helenren) [inline SalesforceSDK/Universal/Upgrade/SDKUpgradeManager.cs]: sure, will add unit test

Comment (helenren) [inline SalesforceSDK/Universal/Upgrade/SDKUpgradeManager.cs]: tests are added.

**Comment by bhariharan** [comment]
👍

**Pull request merged**